### ISSUE TITLE
Slow Inference with RTX 3070

### ISSUE THREAD

**Issue by MYahya3** [Zyphra/Zonos#169]
I am testing Zonos Text-to-Speech with RTX 3070 GPU and it takes around 1.77 seconds to generate 1 second of audio. Is there any way to decrease it because I want ms processing time for audio generation

**Comment by HarleyVader** [comment]
3060 Ti V2 runs. 
needs some optimization but it feels nice.

Not NSFW but its +18. you where warned. 
[melkanea zonos prototype](https://brandynette.xxx/brandynette/melkanea-zonos-prototype/)

sadly i got this beauty

The error message indicates that there is an out-of-bounds index access in a CUDA kernel, specifically in the IndexKernel.cu file. This is likely caused by an invalid index being used in a tensor operation. The traceback points to the following sequence of function calls leading to the error:'

`class Attention(nn.Module):
    def forward(self, x: torch.Tensor, inference_params: InferenceParams, freqs_cis: torch.Tensor) -> torch.Tensor:
        batch_size, seqlen, _ = x.shape

        q_size = self.num_heads * self.head_dim
        kv_size = self.num_heads_kv * self.head_dim
        q, k, v = self.in_proj(x).split([q_size, kv_size, kv_size], dim=-1)
        # ...`

The issue is likely due to the shape of the input tensor [x](vscode-file://vscode-app/c:/Users/urukk/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) not matching the expected dimensions for the projection. This can happen if the input tensor has an incorrect size or if there is a mismatch in the dimensions during the tensor operations.

according to github copilot

**Comment by MYahya3** [comment]
Hi, Thanks for your feedback. Kokoru model is working good in case of Latency even with limited features
